ruby_connectors/lib/ruby_connectors/duckdb_connector/writer.rb

```
@@ -0,0 +1,181 @@
+          raise WriteError, "Failed to write data: #{e.message}"
+        ensure
+          # Make sure appender is closed
+          if appender
+            begin
+              appender.close
+            rescue
+              nil
+            end
+          end
+        end
+      end
```

rubocop (1.69.1)

[convention] Method has too many lines. [21/15]

(Metrics/MethodLength)